pkg/apis/storage/types.go

```diff
@@ -310,6 +327,28 @@ type CSIDriverSpec struct {
 	VolumeLifecycleModes []VolumeLifecycleMode
 }
 
+// SupportsFSGroup specifies if a CSI Driver supports modifying
+// volume ownership and permissions of the volume to be mounted.
+// More modes may be added in the future.
+type SupportsFSGroup string
+
+const (
+	// Evaluate indicates that each volume should be examined
+	// to determine if the volume ownership and permissions
+	// should be modified.
+	EvaluateSupportsFSGroup SupportsFSGroup = "Evaluate"
+
+	// Always indicates that the volume ownership and permissions will
+	// be modified each time the volume is mounted. Volumes
+	// created by this CSI Driver will have their filesystem updated
+	// to match the defined fsGroup on each mount.
+	AlwaysSupportsFSGroup SupportsFSGroup = "Always"
```

Comment:
I think `Always` is not a good word to represent this. For example even if a CSIDriver says `Always` here, this setting could be overridden in pod via `FSGroupChangePolicy` and hence always does not mean fsGroup change will be applied on each mount.

What a CSI driver should be more concerned about is - whether it supports fsGroup or not. So something like - `FSGroupPolicy` with 3 values may be - `Evaulate, Supports, Unsupported` ?

Reply:
I think this sounds good. I'll make these changes tomorrow morning and push them.